debezium-server/debezium-server-redis/src/test/java/io/debezium/server/redis/RedisStreamIT.java

```diff
@@ -61,15 +74,111 @@ void connectorCompleted(@Observes ConnectorCompletedEvent event) throws Exceptio
+         */
+
+        final int MESSAGE_COUNT = 5;
+        final String STREAM_NAME = "testc.inventory.redis_test";
+        Testing.print("Pausing container");
+        RedisTestResourceLifecycleManager.pause();
+        final PostgresConnection connection = getPostgresConnection();
+        Testing.print("Creating new redis_test table and inserting 5 records to it");
+        connection.execute(
+                "CREATE TABLE inventory.redis_test (id INT PRIMARY KEY)",
+                "INSERT INTO inventory.redis_test VALUES (1)",
+                "INSERT INTO inventory.redis_test VALUES (2)",
+                "INSERT INTO inventory.redis_test VALUES (3)",
+                "INSERT INTO inventory.redis_test VALUES (4)",
+                "INSERT INTO inventory.redis_test VALUES (5)");
+        connection.close();
+
+        Testing.print("Sleeping for 5 seconds to simulate no connection errors");
+        Thread.sleep(5000);
+        Testing.print("Unpausing container");
+        RedisTestResourceLifecycleManager.unpause();
+
+        final List<StreamEntry> entries = new ArrayList<>();
+        Awaitility.await().atMost(Duration.ofSeconds(RedisTestConfigSource.waitForSeconds())).until(() -> {
+            final List<StreamEntry> response = jedis.xrange(STREAM_NAME, null, null, MESSAGE_COUNT);
+            entries.clear();
+            entries.addAll(response);
+            return entries.size() >= MESSAGE_COUNT;
+        });
+        Testing.print("Entries in " + STREAM_NAME + ":" + entries.size());
+        assertTrue("Redis Connection Test Failed", entries.size() == MESSAGE_COUNT);
+    }
+
+    @Test
+    @FixFor("DBZ-4510")
+    public void testRedisOOMRetry() throws Exception {
+        /*
+         * Test retry mechanism when encountering Redis Out of Memory:
+         * 1. Simulate a Redis OOM by setting its max memory to 1M.
+         * 2. Create a new table named redis_test2 in PostgreSQL and insert 1000 records to it.
+         * 3. As result, after inserting ~22 records, Redis runs OOM.
+         * 4. Sleep for additional 5 seconds to ensure the Sink is retrying.
+         * 5. Revert max memory setting so Redis is no longer in OOM and make sure all 100 records have been streamed successfully.
+         */
+
+        final int MESSAGE_COUNT = 100;
+        final String STREAM_NAME = "testc.inventory.redis_test2";
+        Testing.print("Setting Redis' maxmemory to 1M");
+        jedis.configSet("maxmemory", "1M");
+        final PostgresConnection connection = getPostgresConnection();
+        Testing.print("Creating new table redis_test2 and inserting 100 records to it");
+        connection.execute(
+                "CREATE TABLE inventory.redis_test2 (id varchar(100) PRIMARY KEY, first_name varchar(100), last_name varchar(100))",
+                "insert into inventory.redis_test2 (id,first_name,last_name) select left(i::text, 10), random()::text, random()::text from generate_series(1,100) s(i)");
```

Comment:
Can you reference `MESSAGE_COUNT` from above to make it more explicit.

Reply:
Yes, makes sense, done.